### PR TITLE
Add logging controller logs on signature operation stages

### PR DESCRIPTION
**Pull request by OGPoyraz**
## Explanation

This PR aims to add `LoggerController:add` calls to messenger on different signature operation stages. (`proposed`, `rejected`, `signed`)

## References

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1320

### `@metamask/signature-controller`

- **ADDED**: Adds LoggingController calls via messenger on signature operation stages

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/signature-controller/src/SignatureController.test.ts [hunk 1]
```diff
@@ -928,4 +937,77 @@ describe('SignatureController', () => {
...
+
+      expect(messengerMock.call).toHaveBeenNthCalledWith(
+        1,
+        'LoggingController:add',
+        {
+          type: LogType.EthSignLog,
+          data: {
+            signingMethod: SigningMethod.EthSignTypedData,
+            stage: SigningStage.Proposed,
+            signingData: expect.anything(),
```
Comment (matthewwalsh0) [inline packages/signature-controller/src/SignatureController.test.ts:958]: We can still use the `expect.any<>`  function but is it worth asserting the standard fields of the data for the sake of safety?

File: packages/signature-controller/src/SignatureController.ts [hunk 2]
```diff
@@ -364,9 +373,16 @@ export class SignatureController extends BaseControllerV2<
     version: string,
     signingOpts: TypedMessageSigningOptions,
   ): Promise<string> {
+    let signTypeForLogger = SigningMethod.EthSignTypedData;
```
Comment (matthewwalsh0) [inline packages/signature-controller/src/SignatureController.ts:376]: Really minor, but is it worth using a nested ternary here or a tiny helper function?

File: packages/signature-controller/src/SignatureController.test.ts (hunk 1 above)
Comment (OGPoyraz) [reply packages/signature-controller/src/SignatureController.test.ts]: Good call, it's worth asserting couple of fields, fixed in https://github.com/MetaMask/core/pull/1692/commits/f67bcbe6345a57cce5b94bf6401142340f9cf959

File: packages/signature-controller/src/SignatureController.ts (hunk 2 above)
Comment (OGPoyraz) [reply packages/signature-controller/src/SignatureController.ts]: I've made it on helper function, nested ternary makes TS to complain.
Fixed in https://github.com/MetaMask/core/pull/1692/commits/95219fbdd86483e0a5830fb89e6de7f367b47b8b

**Review by vinistevam — APPROVED** [review]

**Pull request merged**